### PR TITLE
Patch storagement and apply

### PR DESCRIPTION
**Pull request by edisonhello**
#118

### CODE REVIEW

File: dbux-projects/src/dataLib/TestRun.js [hunk 1]
```diff
@@ -10,6 +10,23 @@ async function newTestRun(bug, result) {
...
+    createAt: new Date(),
+    nFailedTests: -1,
+    timer: undefined,
+    patch: patchString,
+  };
+}
+
+function isTestRunOfBug(obj, bug) {
+  return obj.projectName === bug.project.name && obj.bugId === bug.id;
+}
```
Comment (MiccWan) [inline dbux-projects/src/dataLib/TestRun.js:26]: maybe rename `obj` with `testRun`?

File: dbux-projects/src/dataLib/progressLog.js [hunk 2]
```diff
@@ -24,6 +23,27 @@ async function processBugResult(progressLog, bug, result) {
   saveProgressLog(progressLog);
 }
 
+function processUnfinishTestRun(progressLog, bug, patchString) {
```
Comment (MiccWan) [inline dbux-projects/src/dataLib/progressLog.js:26]: typo processUnfinish**ed**TestRun?

Comment (MiccWan) [reply dbux-projects/src/dataLib/progressLog.js:26]: maybe move this function to `projectsManager`?

**Comment by Domiii** [comment]
@edisonhello 

1. set `cwd` when running any `git` command in `projectPath`
2. add `isCorrectGitRepository` before any `git` call

**Pull request merged**